### PR TITLE
TSCBasic: deprecate `localFileSystem` setter

### PR DESCRIPTION
**Pull request by MaxDesiatov**
This global should never be mutable, especially when using concurrency.

### CODE REVIEW

File: Sources/TSCBasic/FileSystem.swift [hunk 1]
```diff
@@ -1159,8 +1159,19 @@ public class RerootedFileSystemView: FileSystem {
     }
 }
 
+private var _localFileSystem: some FileSystem = LocalFileSystem()
```
Comment (MaxDesiatov) [inline Sources/TSCBasic/FileSystem.swift:1162]: Temporarily introduced for `localFileSystem`'s explicit deprecated setter body to make sense.

**Comment by MaxDesiatov** [comment]
@swift-ci test

**Comment by MaxDesiatov** [comment]
@swift-ci test

**Comment by MaxDesiatov** [comment]
@swift-ci test windows

**Review by neonichu — APPROVED** [review]

**Review by tomerd — APPROVED** [review]

File: Sources/TSCBasic/FileSystem.swift [hunk 2]
```diff
@@ -1159,8 +1159,19 @@ public class RerootedFileSystemView: FileSystem {
...
+private var _localFileSystem: any FileSystem = LocalFileSystem()
+
 /// Public access to the local FS proxy.
-public var localFileSystem: FileSystem = LocalFileSystem()
+public var localFileSystem: any FileSystem {
+    get {
+         return _localFileSystem
+    }
+
+    @available(*, deprecated, message: "This global should never be mutable and is supposed to be read-only")
```
Comment (tomerd) [inline Sources/TSCBasic/FileSystem.swift:1170]: add a date here in a comment so we know when this was first deprecated and get delete once enough time has passed

**Comment by MaxDesiatov** [comment]
@swift-ci test

**Comment by MaxDesiatov** [comment]
@swift-ci test windows

**Comment by MaxDesiatov** [comment]
@swift-ci test windows

**Comment by MaxDesiatov** [comment]
@swift-ci test windows

**Comment by MaxDesiatov** [comment]
@swift-ci test windows

**Pull request merged**